issue_10/script.js

```diff
@@ -0,0 +1,58 @@
+function startLoading() {
+    const loadingImg = document.createElement('img');
+    loadingImg.src = 'loading-circle.gif';
+    loadingImg.alt = 'ローディング画像';
+    document.getElementById('js-loading').appendChild(loadingImg);
+}
+
+function stopLoading() {
+    document.getElementById('js-loading').remove();
+}
+
+function createLists(resolvedData) {
+    const ul = document.getElementById('js-ul');
+    const fragment = document.createDocumentFragment();
+    for (const item of resolvedData) {
+        const li = document.createElement('li');
+        const p = document.createElement('p');
+        const img = document.createElement('img');
+
+        p.textContent = item.text;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(p).appendChild(img);
+    }
+    ul.appendChild(fragment);
+}
+
+function getListData() {
+    return new Promise((resolve) => {
+        const lists = [{
+            img: "1.png",
+            alt: "画像1",
+            text: "ブックマーク"
+        }, {
+            img: "2.png",
+            alt: "画像2",
+            text: "メッセージ"
+        }];
+        setTimeout(() => {
+            resolve(lists);
+        }, 3000);
+    });
+}
+
+async function showList() {
+    startLoading();
+    try {
+        const listData = await getListData();
+        createLists(listData);
```

Comment:
In this case, only 'getData' should be implemented in the ‘try', because if an error occurs, it is impossible to determine where the error occurred. (I was reviewed like this before!)

Reply:
Thanks for your suggestion:sparkles:
It is indeed difficult to identify when an error occurs.
I have modified it to define another function and execute createList there.
https://github.com/yoshikawa-akr/js-lessons/pull/10/commits/059005bc219206e7c56da0639fc5e839b8726f62

Reply:
I think it's good!!:thumbsup: